pf4j-spring/pom.xml

```diff
@@ -33,6 +33,12 @@
             <version>${spring.version}</version>
         </dependency>
 
+        <dependency>
```

Comment:
This library is built by default with Java 8.
I suggest you to put this dependency as runtime dependency,  in your application.
Other variant is to use profile (Java 9+) dependency (for details see [this](https://stackoverflow.com/a/167284))